subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -229,8 +229,11 @@ static void tls_debug(void *ctx, int level, const char *file,
 		}
 	}
 
-	NET_DBG("%s:%04d: |%d| %s", basename, line, level,
-		log_strdup(str));
+	char *log = log_strdup(str);
+
+	/* Remove extra EOL to improve log readability. */
+	log[strlen(str) - 1] = '\0';
```

Comment:
I don't think this is safe... In certain configurations `log_strdup()` will simply forward the pointer w/o making a copy (https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/logging/log.h#L293), just dropping the `const` specifier...

Reply:
Yeah, that's an iffy change.  Of course, if it does forward the pointer w/o making a copy, then `log_strdup()` is also doing something questionable by returning a non-const pointer.

I just changed it to skip the final character without modifying the string.

Reply:
> Would be good to get a word from mbed TLS fork maintainers, whether I'd be even allowed to cherry-pick certain mbed TLS functionalities from upstream.

No doubt.  I was expecting resistance on that part of this PR, but felt it was better to get it out there for discussion.

Reply:
I made the debug output change a separate PR -- I'll pull it out of this when the separate one merges